source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -135,15 +132,20 @@ void AuthenticatorImpl::startVerify() {
     }
 
     // Check the issuer is configured or not.
-    jwks_data_ = jwks_cache_.findByIssuer(jwt_.iss_);
+    if (provider_) {
+      jwks_data_ = jwks_cache_.findByProvider(provider_.value());
+    } else {
+      jwks_data_ = jwks_cache_.findByIssuer(jwt_.iss_);
+    }
     // isIssuerSpecified() check already make sure the issuer is in the cache.
     ASSERT(jwks_data_ != nullptr);
 
-    // Check if audience is allowed
-    bool allowed = audiences_ ? audiences_->areAudiencesAllowed(jwt_.audiences_)
-                              : jwks_data_->areAudiencesAllowed(jwt_.audiences_);
+    const AudienceChecker& audience_checker =
```

Comment:
I feel this audience_checker_supplier is over-kill.   It will be much simpler to just provide a
const CheckAudience* to Auth constructor and ProviderWithAudience will provide a valid one.
In the checking,  if it is valid, use it to check,  otherwise just use  the one from jwks_data_